libs/cli/src/main/java/org/elasticsearch/cli/Terminal.java

```diff
@@ -216,5 +245,58 @@ public String readText(String text) {
         public char[] readSecret(String text) {
             return readText(text).toCharArray();
         }
+
+        @Override
+        public char[] readSecret(String text, int maxLength) {
+            getErrorWriter().println(text);
```

Comment:
Shouldn't this message be to stdout, not stderr?

Reply:
This is intentional. Consider the case of wanting to grep for the presence of particular settings in the keystore.

```
bin/elasticsearch-keystore list | grep 'xpack'
```

If we prompt on standard input, the prompt will be passed to grep and the user won't see it. Even if the user knows to enter a password, the next tool in the pipe will still receive the prompt as if it's output data.

The bash builtin `read` function does the same thing with its prompts:

```
       read [-ers] [-a aname] [-d delim] [-i text] [-n nchars] [-N nchars] [-p prompt] [-t time-
       out] [-u fd] [name ...]
[...skipping lots of text...]
              -p prompt
                     Display  prompt  on  standard  error,  without  a  trailing newline, before
                     attempting to read any input.  The prompt is displayed  only  if  input  is
                     coming from a terminal.
```